pkg/api/v1alpha1/cluster_types.go

```diff
@@ -569,6 +570,12 @@ const (
 	Kube124 KubernetesVersion = "1.24"
 )
 
+// KubeVersionToValidSemver converts kube version to semver for comparisons.
+func KubeVersionToValidSemver(kubeVersion KubernetesVersion) (*semver.Version, error) {
```

Comment:
This shouldn't be in the cluster_types.go file I think. Sounds like a utility function based on when we need to use, because this gives the impression that `.0` is the version used in the cluster.

Reply:
moved it to semver utils pkg